src/main/java/bio/terra/service/filedata/flight/ingest/IngestFileDirectoryStep.java

```diff
@@ -51,13 +67,27 @@ public StepResult doStep(FlightContext context) {
                     .isFileRef(true)
                     .path(fireStoreUtils.getDirectoryPath(loadModel.getTargetPath()))
                     .name(fireStoreUtils.getName(loadModel.getTargetPath()))
-                    .datasetId(datasetId);
+                    .datasetId(datasetId)
+                    .loadTag(loadModel.getLoadTag());
                 fileDao.createDirectoryEntry(dataset, newEntry);
             } else {
-                if (!StringUtils.equals(existingEntry.getFileId(), fileId)) {
-                    // Exists, but is not our file!
+                if (!StringUtils.equals(existingEntry.getLoadTag(), loadModel.getLoadTag())) {
+                    // (a) Exists and is not our file
                     throw new FileAlreadyExistsException("Path already exists: " + targetPath);
                 }
+
+                // (b) or (c) Load tags match - check file ids
+                if (!StringUtils.equals(existingEntry.getFileId(), fileId)) {
+                    // (c) We are in a re-run of a load job. Try to get the file entry.
+                    fileId = existingEntry.getFileId();
+                    workingMap.put(FileMapKeys.FILE_ID, fileId);
+                    FireStoreFile fileEntry = fileDao.lookupFile(dataset, fileId);
+                    if (fileEntry != null) {
+                        // (c)(i) We successfully loaded this file already
+                        workingMap.put(FileMapKeys.LOAD_COMPLETED, true);
+                    }
```

Comment:
Can you add a comment here about `(c)(ii)` leaving load_completed `false`? I like how you've "linked" back to the comments above.